### PR TITLE
Notifications

### LINKED ISSUE #816: Notifications
**Is your feature request related to a problem? Please describe.**
We often need to display some sort of notification to the user and we currently have no way to put them.
e.g your upload of this image failed

**Describe the solution you'd like**
Include a Notification icon where notifications can be stored and where upcoming notification would be shown.

**Describe alternatives you've considered**
Implement notifications in another way.

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Description
This PR adds dismissable, temporary notifications. Every notification can have actions to execute.

### Steps

- [x] Added implementation
- [ ] Added / updated tests
- [ ] Added / updated documentation
- [ ] Extended changelog
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
Closes #816

### CODE REVIEW

File: src/components/notifications/ui-notification-toast.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import { dismissUiNotification } from '../../redux/ui-notifications/methods'
+
+const stepsPerSecond = 10
+
+export interface UiNotificationProps extends UiNotification {
+  notificationId: number
+}
+
+export const UiNotificationToast: React.FC<UiNotificationProps> = ({ title, content, date, icon, dismissed, notificationId, durationInSecond, buttons }) => {
+  const [eta, setETA] = useState<number>()
```
Comment (ErikMichelson) [inline src/components/notifications/ui-notification-toast.tsx:22]: When using `eta` in all lowercase, only the E should be upper-case in the setter.

```suggestion
  const [eta, setEta] = useState<number>()
```

File: src/components/notifications/ui-notification-toast.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+import React, { useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { Button, ProgressBar, Toast } from 'react-bootstrap'
+import { UiNotification } from '../../redux/ui-notifications/types'
+import { ForkAwesomeIcon } from '../common/fork-awesome/fork-awesome-icon'
+import { ShowIf } from '../common/show-if/show-if'
+import { IconName } from '../common/fork-awesome/types'
+import { dismissUiNotification } from '../../redux/ui-notifications/methods'
+
+const stepsPerSecond = 10
```
Comment (ErikMichelson) [inline src/components/notifications/ui-notification-toast.tsx:15]: Static constant should be all uppercase.

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: src/components/notifications/ui-notification-toast.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/notifications/ui-notification-toast.tsx]: done

File: src/components/notifications/ui-notification-toast.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/notifications/ui-notification-toast.tsx]: done

File: src/components/editor-page/use-notification-test.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/*
```
Comment (DerMolly) [inline src/components/editor-page/use-notification-test.tsx:1]: Should this be part of this PR?

Comment (ErikMichelson) [reply src/components/editor-page/use-notification-test.tsx:1]: Yes. It will be replaced as soon as we have real notification integrations.

Comment (mrdrogdrog) [reply src/components/editor-page/use-notification-test.tsx:1]: What he says

**Review by DerMolly — APPROVED** [review]

**Pull request merged**